tests/test__collection_api.py

```diff
@@ -5983,3 +5983,34 @@ def test_elem_match(self):
         ids = set(doc['_id'] for doc in self.db.collection.find(
             {'arr': {'$elemMatch': {'$lt': 10, '$gt': 4}}}, {'_id': 1}))
         self.assertEqual({1, 2}, ids)
+
+    def test_list_collection_names_filter(self):
+        self.db.create_collection('aggregator')
+        for day in range(10):
+            new_date = datetime.now() - timedelta(day)
```

Comment:
I suggest you compute datetime.now() only once at the top of your test: if the testing happens exactly at a date change the you would make sure that it doesn't impact the test.